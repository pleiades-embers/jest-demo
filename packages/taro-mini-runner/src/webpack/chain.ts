import { ICopyOptions } from '@tarojs/taro/types/compile'
import { IOption } from '../utils/types'
import CopyPlugin from 'copy-webpack-plugin'
import { partial } from 'lodash'
import path from 'path'

const getPlugin = (plugin:any, args:IOption[]) => {
  return {
    plugin, args
  }
}

export const getCopyWebpackPlugin = ({
  copy,
  appPath
}): {
  copy: ICopyOptions
  appPath: string
} => {
  const args = [
    copy.patterns.map(({ from, to, ...extra }) => {
      return {
        from,
        to: path.resolve(appPath, to),
        context: appPath,
        ...extra
      }
    }),
    copy.options
  ]

  return partial(getPlugin, CopyPlugin)(args)
}
