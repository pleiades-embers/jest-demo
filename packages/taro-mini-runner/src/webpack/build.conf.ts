import { IBuildConfig } from '../utils/types'
import { PLATFORMS, taroJsComponents } from '@tarojs/helper'
import { componentConfig } from '../template/component'
import { getCopyWebpackPlugin } from './chain'

export default (appPath: string, mode, config: Partial<IBuildConfig>): any => {
  console.log(appPath, mode, config)
  const {
    buildAdapter = PLATFORMS.WEAPP,
    alias = {},
    // entry = {},
    // output = {},
    // fileType = {
    //   style: '.wxss',
    //   config: '.json',
    //   script: '.js',
    //   templ: '.wxml'
    // }
    taroComponentsPath,
    env = {}
  } = config
  // build 时修改 组件配置
  config.modifyComponentConfig?.(componentConfig, config)

  const { copy } = config
  const plugin:any = []
  if (copy) {
    plugin.copyWebpackPlugin = getCopyWebpackPlugin({ copy, appPath })
  }

  alias[taroJsComponents + '$'] = taroComponentsPath || `${taroJsComponents}/mini`

  //缓存 环境信息
  env.TARO_ENV = JSON.stringify(buildAdapter)
}
