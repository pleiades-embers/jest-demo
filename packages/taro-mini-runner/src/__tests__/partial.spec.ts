import { partial } from 'lodash'
import { getCopyWebpackPlugin } from '../webpack/chain'
describe('partial', () => {
  test('should partial', async () => {
    const greet = function (greeting, name) {
      return greeting + ' ' + name
    }

    const sayHelloTo = partial(greet, 'hello')

    // 使用了占位符。
    const greetFred = partial(greet, 'fred')

    expect(sayHelloTo('fred')).toMatchInlineSnapshot(`"hello fred"`)
    expect(greetFred('hi')).toMatchInlineSnapshot(`"fred hi"`)
    const copy = {}
    expect(getCopyWebpackPlugin({ copy, appPath: 'react' })).toMatchInlineSnapshot()
  })
})
