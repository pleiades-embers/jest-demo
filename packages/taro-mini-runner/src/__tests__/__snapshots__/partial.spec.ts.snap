// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`partial should partial 1`] = `"hello fred"`;

exports[`partial should partial 2`] = `"fred hi"`;
